src/components/LoginStatus.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { User } from "../slices/userSlice";
+import { connect } from "react-redux";
+import { getUserFullName, getIsLoggedIn } from "../slices/userSlice";
+
+type OwnProps = {
+  name?: string;
+  isLoggedIn: boolean;
+  loginURI: string;
+  logoutURI: string;
+};
+
+type ComponentSettings = {
+  linkURI: string;
+  linkText: string;
+  linkClass: string;
+  containerClass: string;
+  nameComponent?: JSX.Element;
+};
+
+const getSettings = (props: OwnProps): ComponentSettings => {
+  const { isLoggedIn, logoutURI, loginURI, name } = props;
+  if (isLoggedIn) {
+    return {
+      linkURI: logoutURI,
+      linkText: "Logout",
+      linkClass: "login-status__link--small",
+      containerClass: "login-status--small",
+      nameComponent: <div className="login-status__name">{name}</div>
+    };
+  } else {
+    return {
+      linkURI: loginURI,
+      linkText: "Login",
+      linkClass: "login-status__link",
+      containerClass: "login-status"
+    };
+  }
+};
+
+export const LoginStatus: React.FC<OwnProps> = props => {
+  const {
+    containerClass,
+    nameComponent,
+    linkClass,
+    linkURI,
+    linkText
+  } = getSettings(props);

```

Comment:
While I see the benefit of moving the application state to UI mapping to a separate function, in my opinion this is a trade-off, where the code readability gets worse. When I look at the returned JSX, I have to scroll back and forth to the `getSettings` if I want to see the logic behind. Including the logic in the markup might seem a bit odd at first, but it's decreasing the time required to understand the code.